addons/stock_picking_batch/models/stock_picking_batch.py

```diff
@@ -136,14 +136,15 @@ def _set_move_line_ids(self):
     # -------------------------------------------------------------------------
     # CRUD
     # -------------------------------------------------------------------------
-    @api.model
-    def create(self, vals):
-        if vals.get('name', '/') == '/':
-            if vals.get('is_wave'):
-                vals['name'] = self.env['ir.sequence'].next_by_code('picking.wave') or '/'
-            else:
-                vals['name'] = self.env['ir.sequence'].next_by_code('picking.batch') or '/'
-        return super().create(vals)
+    @api.model_create_multi
+    def create(self, vals_list):
+        for vals in vals_list:
+            if vals.get('name', '/') == '/':
+                if vals.get('is_wave'):
+                    vals['name'] = self.env['ir.sequence'].next_by_code('picking.wave') or '/'
+                else:
+                    vals['name'] = self.env['ir.sequence'].next_by_code('picking.batch') or '/'
```

Comment:
```suggestion
                company_id = vals.get('company_id', self.env.company.id)
                if vals.get('is_wave'):
                    vals['name'] = self.env['ir.sequence'].with_company(company_id).next_by_code('picking.wave') or '/'
                else:
                    vals['name'] = self.env['ir.sequence'].with_company(company_id).next_by_code('picking.batch') or '/'
```
AFAIK, sequences are company dependent.